Tests/XMLCoderTests/Minimal/UnkeyedIntTests.swift

```diff
@@ -0,0 +1,261 @@
+}
+
+extension IntegerArrayContainer {
+    var intValue: [Int] {
+        return value.map { Int($0) }
+    }
+}
+
+enum CodingKeys: CodingKey {
+    case value
+}
+
+private func decodeArray<T>(
+    _ decoder: Decoder,
+    decode: (inout UnkeyedDecodingContainer) throws -> T
+) throws -> [T] {
+    let keyedContainer = try decoder.container(keyedBy: CodingKeys.self)
+    var container = try keyedContainer.nestedUnkeyedContainer(forKey: .value)
+
+    var decoded = [T]()
+    var finished = false
+
+    while !finished {
+        do {
+            decoded.append(try decode(&container))
+        } catch DecodingError.valueNotFound {
+            finished = true
+        } catch {
+            throw error
+        }
+    }
+
+    return decoded
+}
+
+struct UnkeyedContainerInt8: Codable, Equatable, IntegerArrayContainer {
+    let value: [Int8]
+
+    init(from decoder: Decoder) throws {
+        value = try decodeArray(decoder) {
+            try $0.decode(Int8.self)
+        }
+    }
+}
+
+struct UnkeyedContainerInt16: Codable, Equatable, IntegerArrayContainer {
+    let value: [Int16]
+
+    init(from decoder: Decoder) throws {
+        value = try decodeArray(decoder) {
+            try $0.decode(Int16.self)
+        }
+    }
+}
+
+struct UnkeyedContainerInt32: Codable, Equatable, IntegerArrayContainer {
+    let value: [Int32]
+
+    init(from decoder: Decoder) throws {
+        value = try decodeArray(decoder) {
+            try $0.decode(Int32.self)
+        }
+    }
+}
+
+struct UnkeyedContainerInt64: Codable, Equatable, IntegerArrayContainer {
+    let value: [Int64]
+
+    init(from decoder: Decoder) throws {
+        value = try decodeArray(decoder) {
+            try $0.decode(Int64.self)
+        }
+    }
+}
+
+struct UnkeyedContainerUInt8: Codable, Equatable, IntegerArrayContainer {
+    let value: [UInt8]
+
+    init(from decoder: Decoder) throws {
+        value = try decodeArray(decoder) {
+            try $0.decode(UInt8.self)
+        }
+    }
+}
+
+struct UnkeyedContainerUInt16: Codable, Equatable, IntegerArrayContainer {
+    let value: [UInt16]
+
+    init(from decoder: Decoder) throws {
+        value = try decodeArray(decoder) {
+            try $0.decode(UInt16.self)
+        }
+    }
+}
+
+struct UnkeyedContainerUInt32: Codable, Equatable, IntegerArrayContainer {
+    let value: [UInt32]
+
+    init(from decoder: Decoder) throws {
+        value = try decodeArray(decoder) {
+            try $0.decode(UInt32.self)
+        }
+    }
+}
+
+struct UnkeyedContainerUInt64: Codable, Equatable, IntegerArrayContainer {
+    let value: [UInt64]
+
+    init(from decoder: Decoder) throws {
+        value = try decodeArray(decoder) {
+            try $0.decode(UInt64.self)
+        }
+    }
+}
+
+let xmlString =
+    """
+    <container>
+        <value>42</value>
+        <value>43</value>
+        <value>44</value>
+    </container>
+    """
+
+class UnkeyedIntTests: XCTestCase {
+    func testInt8() throws {
```

Comment:
could we create a new `testIntArray` generic function here and call that function with a metatype argument similar to what is available in `KeyedIntTests`?